src/openzaak/components/besluiten/api/viewsets.py

```diff
@@ -355,3 +366,101 @@ class BesluitAuditTrailViewSet(AuditTrailViewSet):
 
     main_resource_lookup_field = "besluit_uuid"
     permission_classes = (AuthRequired,)
+
+
+@extend_schema(
+    summary="Verwerk een besluit",
+    description=mark_experimental(
+        "Maak een Besluit aan met en een of meerdere BesluitInformatieObject(en) aan om documenten direct aan een besluit te linken."
+    ),
+)
+class BesluitVerwerkenViewSet(
+    viewsets.ViewSet,
+    MultipleNotificationMixin,
+    ClosedZaakMixin,
+    MultipleAuditTrailMixin,
+):
+    serializer_class = BesluitVerwerkenSerializer
+    permission_classes = (BesluitVerwerkenAuthRequired,)
+
+    required_scopes = {"create": SCOPE_BESLUITEN_AANMAKEN}
+
+    viewset_classes = {
+        "besluit": "openzaak.components.besluiten.api.viewsets.BesluitViewSet",
+    }
+
+    notification_fields = {
+        "besluit": {
+            "notifications_kanaal": KANAAL_BESLUITEN,
+            "model": Besluit,
+        },
+        "besluitinformatieobjecten": {
+            "notifications_kanaal": KANAAL_BESLUITEN,
+            "model": BesluitInformatieObject,
+        },
+    }
+
+    @transaction.atomic
+    def create(self, request, *args, **kwargs):
+        serializer = self.serializer_class(
+            data=request.data, context={"request": request}
+        )
+        serializer.is_valid(raise_exception=True)
+
+        self.perform_create(serializer)
+
+        response = Response(serializer.data, status=status.HTTP_201_CREATED)
+
+        self.create_audittrail(
+            response.status_code,
+            CommonResourceAction.create,
+            version_before_edit=None,
+            version_after_edit=serializer.data["besluit"],
+            unique_representation=serializer.instance[
+                "besluit"
+            ].unique_representation(),
+            audit=AUDIT_BRC,
+            basename="besluit",
```

Comment:
You could maybe pull this and the other `basename` below from `serializer.instance[...]._model_name` (forgot what it was exactly)?